tests/helpers/test-errors-on-directive-linking.js

```diff
@@ -0,0 +1,122 @@
+/* jshint es3: false, esnext: true */
+/* global jasmine, beforeEach, describe, it, expect, module, inject */
+
+(function () {
+
+    var helpers = window._helpers = (window._helpers || {});
+
+    helpers.testErrorsOnDirectiveLinking = function (moduleName, options) {
+
+        'use strict';
+
+        if (!options) {
+            throw new Error('options required');
+        }
+
+        if (!options.directives) {
+            throw new Error('directives option required');
+        }
+        if (!options.factory) {
+            throw new Error('factory options required');
+        }
+
+        var factoryName = options.factory.name;
+        var factoryMethods = options.factory.methods;
+
+
+        describe(moduleName + ': getting fields from OPTIONS request', function () {
```

Comment:
In other words, rather than writing DRY tests it would be better to DRY the code to a service that does something like:
```
(response) =>  response.data.actions && (response.data.actions.POST || response.data.actions.PUT)
```